traffic_ops/app/lib/API/CachegroupFallback.pm

```diff
@@ -0,0 +1,208 @@
+package API::CachegroupFallback;
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+#
+# a note about locations and cachegroups. This used to be "Location", before we had physical locations in 12M. Very confusing.
+# What used to be called a location is now called a "cache group" and location is now a physical address, not a group of caches working together.
+#
+
+# JvD Note: you always want to put Utils as the first use. Sh*t don't work if it's after the Mojo lines.
+use UI::Utils;
+use Mojo::Base 'Mojolicious::Controller';
+use Data::Dumper;
+use JSON;
+use MojoPlugins::Response;
+use Validate::Tiny ':all';
+
+sub delete {
+	my $self = shift;
+	my $cache_id = $self->param('cacheGroupId');
+	my $params = $self->req->json;
+
+	if ( !&is_oper($self) ) {
+		return $self->forbidden();
+	}
+
+	#only integers
+	if ( $cache_id !~ /^\d+?$/ ) {
+		&log( $self, "No such Cachegroup id $cache_id");
+		return $self->not_found();
+	}
+
+	my $cachegroup = $self->db->resultset('Cachegroup')->search( { id => $cache_id } )->single();
+	if ( !defined($cachegroup) ) {
+		&log( $self, "No such Cachegroup $cache_id");
+		return $self->not_found();
+	}
+
+	if ( ($cachegroup->type->name ne "EDGE_LOC") ) {
+		return $self->alert("cachegroup should be type EDGE_LOC.");
+	}
+
+	my $rs_backups = $self->db->resultset('CachegroupFallback')->search( { primary_cg => $cache_id } );
+	if ( ($rs_backups->count > 0) ) {
+		my $del_records = $rs_backups->delete();
+		if ($del_records) {
+			&log( $self, "Backup configuration for cache group $cache_id DELETED", "APICHANGE");
+			return $self->success_message("Backup configuration for cache group $cache_id DELETED");
+		} else {
+			return $self->alert( "Backup configuration for cache group $cache_id DELETED." );
+		}
+	} else {
+		&log( $self, "No backup Cachegroups for $cache_id");
+		return $self->not_found();
+	}
+}
+
+sub show {
```

Comment:
how can i fetch just one row? like this

GET /api/1.2/cachegroup_fallbacks?cacheGroupId=&fallbackId=